engine/class_modules/sc_mage.cpp

```diff
@@ -3513,9 +3520,19 @@ struct arcane_barrage_t final : public dematerialize_spell_t
     // Arcane Charge from the Orb cast increases Barrage damage, but does not change
     // how many targets it hits. Snapshot the buff stacks before executing the Orb.
     snapshot_charges = p()->buffs.arcane_charge->check();
-    if ( rng().roll( snapshot_charges * p()->talents.orb_barrage->effectN( 1 ).percent() ) )
+    if ( rng().roll( snapshot_charges * p()->talents.orb_barrage->effectN( 1 ).percent() ) ) {
       orb_barrage->execute_on_target( target );
 
+      //so i had to repeat this block of code three times (ae echo, orb barrage, sft orb), which is pretty ugly, and i think its preferable to leave it within mage_spell's execute, 
+      //but i don't know how to cleanly refer to those three as theyre considered within the background. please lemme know if there's a way to refer to a specific action being executed within mage_spell.
```

Comment:
There is (see how we handle Scorch/IB in `handle_hot_streak`), but it would be much easier of you just refactored this into a new method (i.e. something like `trigger_intuition`) somewhere in `arcane_mage_spell_t`/`mage_spell_t`.